src/functions.jl

```diff
@@ -951,6 +1019,14 @@ end
 
 # ScalarNonlinearFunction
 
+function Base.convert(::Type{ScalarNonlinearFunction}, x::Real)
+    return ScalarNonlinearFunction(:+, Any[x])
+end
+
+function Base.convert(::Type{ScalarNonlinearFunction}, x::VariableIndex)
+    return ScalarNonlinearFunction(:+, Any[x])
```

Comment:
So I wondered about introducing `:identity` as a special-case nonlinear argument.

Then `VectorNonlinearFunction` would be `Vector{ScalarNonlinearFunction}`, and there wouldn't be this distinction between `scalarize` returning the `Vector{Any}` and `eachscalar` returning an iterator that converted each element to `ScalarNonlinearFunction`.

I guess `identity` is the same as univariate `:+`, but it's more explicit as to what is happening.

Reply:
Yes, I also prefer `identity`. Indeed, I can see the issue that `scalar_type` is `ScalarNonlinearFunction` so `eachscalar` has to convert to `ScalarNonlinearFunction which is inconsistent. Maybe that's a good argument against `Vector{Any}`

Reply:
So I've been experimenting with this. It really doesn't add anything, and only complicates stuff. let's just leave as `:+` for now.